jaxopt/_src/levenberg_marquardt.py

```diff
@@ -326,53 +345,49 @@ def update(self, params, state: NamedTuple, *args, **kwargs) -> base.OptStep:
     # Current value of the loss function F=0.5*||f||^2.
     loss_curr = state.loss
 
+    # TODO: clean up the linear_solver to take matrix directly and then clean up
+    # linear equation solves for materialize_jac=True to call them.
+
     # For geodesic acceleration, we calculate  jtrpp=JT * r",
     # where J is jacobian and r" is second order directional derivative.
     if self.materialize_jac:
       damping_term = state.damping_factor * jnp.identity(params.size)
       # Note that instead of taking the inverse of jtj_corr and multiply that
-      # by state.gradient, we are using `solve`, which uses LU decomposition of
+      # by state.gradient, we prefer to use `jsp.solve`, which uses Cholesky of
       # jtj_corr and uses that to obtain velocity. This has the advantage of
       # lower number of floating point operations and therefore less numerical
       # error which can be helpful for the case of single precision arithmatics.
       jtj_corr = state.jtj + damping_term
-      velocity = jnp.linalg.solve(jtj_corr, state.gradient)
+      if self.solver == 'lu':
+        velocity = jnp.linalg.solve(jtj_corr, state.gradient)
+      elif self.solver == 'qr':
+        q, r = jsp.linalg.qr(jtj_corr)
+        velocity = jsp.linalg.solve_triangular(r, q.T @ state.gradient)
+      elif self.solver == 'inv':
+        velocity = jnp.linalg.solve(jtj_corr, state.gradient)
+      else:
+        velocity = jsp.linalg.solve(jtj_corr, state.gradient, assume_a='pos')
       delta_params = velocity
       if self.geodesic:
         rpp = (state.hess_res @ velocity) @ velocity
         # Note the same as above here that we could use inverse of jtj_corr but
         # chose to use solve for higher performance and lower numerical error.
-        acceleration = jnp.linalg.solve(jtj_corr, state.jt)
+        if self.solver == 'lu':
+          acceleration = jnp.linalg.solve(jtj_corr, state.jt)
```

Comment:
I don't see the difference between this line and line 380. 
Also, generally I would remove 'inv' since it is the least accurate and most costly algorithm. It would be useful if the inverse was kept in memory. We may keep it for now for legacy reasons.